src/puppetlabs/pcp/message.clj

```diff
@@ -96,17 +93,21 @@
 
 (s/defn get-json-data :- s/Any
   "Returns the data from the data frame decoded from json"
-  [message :- Message]
-  (let [data (get-data message)
-        decoded (cheshire/parse-string (bytes->string data) true)]
-    decoded))
+  ([message] (get-json-data message "v2.0"))
+  ([message :- Message
+    version]
+   (case version
+     "v1.0" (-> (get-data message)
+                bytes->string
+                (cheshire/parse-string true))
+     (:data message))))
 
 (s/defn get-json-debug :- s/Any
```

Comment:
I left this in place because it's still used in the decode function and I don't have a sense of our commitments to backward compatibility for these functions. It seems to me like we should still be able to create/interpret/test a v1 message with debug chunks for backward compatibility. If that's not the case then you're right about this and the other functions.